File: frontend/src/app/(dashboard)/advice-consulting/_actions/adviceConsulting.ts
'use server';

import { adviceConsultingDB } from '@/app/_db/advice-consulting';
import { LLMserviceBackEndLog } from '@/app/_types/logs/llm-services';
import { getCurrentUser } from '@/app/_utils/auth';
import { getMessage } from '@/app/_utils/message';
import { useCaseAzureFunctions } from '../../../../../azure-functions';
import { adviceConsultingContainer } from '../../../../../cosmos';

type AdviceConsultingResponse = {
  result: string;
  success: boolean;
  log: LLMserviceBackEndLog<'adviceConsulting'>;
};

type AdviceConsultingErrorResponse = {
  error: string;
};

export async function createAdviceConsulting(
  id: string,
  role: string,
  constraints: string,
  adviceInput: string
): Promise<AdviceConsultingResponse | AdviceConsultingErrorResponse> {
  const user = await getCurrentUser();
  try {
    const response = await useCaseAzureFunctions.sendJson<
      {
        role: string;
        constraints: string;
        adviceInput: string;
      },
      AdviceConsultingResponse
    >('advice-consulting', 'POST', {
      role,
      constraints,
      adviceInput,
    });

    // CosmosDBにログを保存
    await adviceConsultingDB(adviceConsultingContainer, {
      id,
      userId: user.id,
      userName: user.name,
      userEmail: user.email,
      userDepartmentName: user.departmentName,
      role,
      constraints,
      adviceInput,
      createdAt: new Date(),
      outputForm: response.result,
      log: response.log,
    });

    return response;
  } catch (error) {
    console.error('Error creating advice consulting:', error);
    return {
      error:
        error instanceof Error
          ? error.message
          : getMessage('E_F_00110', 'アドバイス（コンサルティング）'),
    };
  }
}

File: backend/orchestrator-prompt/system/advice_consulting.py
# アドバイス(コンサルティング)
def get_advice_consulting_system_message(
    role: str,
    constraints: str,
    adviceInput: str,
) -> str:
    ADVICE_CONSULTING_SYSTEM_MESSAGE = f"""\
    # 役割
    あなたは優秀なコンサルタントです。

    # 指示
    入力欄の制約条件を基に、指定された役割としてわかりやすい具体例を示しながらアドバイスを出力してください。

    ### 入力欄
    役割：{role}
    制約条件：{constraints}
    アドバイスをもらいたいこと：{adviceInput}

    # 出力形式
    - 具体例を含めたわかりやすいアドバイスを提供してください
    - 制約条件に従って出力してください
    - 指定された役割の視点からアドバイスしてください
    - 実践的で行動可能なアドバイスを心がけてください
    """
    return ADVICE_CONSULTING_SYSTEM_MESSAGE


def get_advice_consulting_message(
    role: str,
    constraints: str,
    adviceInput: str,
):
    messages = [
        {
            "role": "user",
            "content": get_advice_consulting_system_message(
                role,
                constraints,
                adviceInput,
            ),
        },
    ]
    return messages
